src/System.Web.OData/OData/Query/Expressions/ExpressionBinderBase.cs

```diff
@@ -1247,6 +1248,33 @@ public virtual Expression BindSingleValueFunctionCallNode(SingleValueFunctionCal
             }
         }
 
+        private Expression BindIif(SingleValueFunctionCallNode node)
+        {
+            Contract.Assert("iif" == node.Name);
+
+            Expression[] arguments = BindArguments(node.Parameters);
+
+            if (QuerySettings.HandleNullPropagation == HandleNullPropagationOption.True)
+            {
+                // we don't have to check if the argument is null inside the function call as we do it already
+                // before calling the function. So remove the redundant null checks.
+                for (int i = 0; i < arguments.Length; i++)
+                {
+                    arguments[i] = RemoveInnerNullPropagation(arguments[i]);
+                }
+            }
+
+            if (arguments[0].Type != typeof(bool))
+            {
+                throw new ODataException(Error.Format(SRResources.FunctionNotSupportedOnEnum, node.Name));
+            }
```

Comment:
This type is already checked during Uri parsing. Please consider using `Contract.Assert(` to comply with other, similar functions.